test/gmp/GeneralMessagePassing.js

```diff
@@ -142,12 +143,18 @@ describe('GeneralMessagePassing', () => {
         tokenA = await mintableCappedERC20Factory.deploy(nameA, symbolA, decimals, capacity).then((d) => d.deployed());
         tokenB = await mintableCappedERC20Factory.deploy(nameB, symbolB, decimals, capacity).then((d) => d.deployed());
 
-        await sourceChainGateway.execute(
-            await getSignedWeightedExecuteInput(await getTokenDeployData(false), [operatorWallet], [1], 1, [operatorWallet]),
-        );
-        await destinationChainGateway.execute(
-            await getSignedWeightedExecuteInput(await getTokenDeployData(true), [operatorWallet], [1], 1, [operatorWallet]),
-        );
+        await sourceChainGateway
+            .execute(
+                await getSignedWeightedExecuteInput(await getTokenDeployData(false), [operatorWallet], [1], 1, [operatorWallet]),
+                getGasOptions(),
```

Comment:
`gateway.execute` skips over failed calls within a loop. this makes it hard to estimate gas, since technically just enough gas can be provided for tx to execute without processing any commands. So, we should pass in `getGasOptions` wherever gateway.execute is used.